core/utils/mailbox.go

```diff
@@ -73,6 +73,17 @@ func (m *Mailbox[T]) Retrieve() (t T, ok bool) {
 	return
 }
 
+func (m *Mailbox[T]) RetrieveAll() []T {
```

Comment:
Order here is pretty important to keep consistent as the internal queue is ordered opposite of what this function returns. Can you add some unit tests to help ensure the inner workings of this function remain as expected? We don't want any future accidents here.

Reply:
Sure np